src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.c

```diff
@@ -242,6 +243,46 @@ void grpc_resolve_address_ares_impl(grpc_exec_ctx *exec_ctx, const char *name,
   r->success = false;
   r->error = GRPC_ERROR_NONE;
   ares_channel *channel = grpc_ares_ev_driver_get_channel(r->ev_driver);
+
+  // If dns_server is specified, use it.
+  if (dns_server != NULL) {
+    gpr_log(GPR_INFO, "Using DNS server %s", dns_server);
+    struct sockaddr_storage sockaddr;
+    if (grpc_parse_ipv4_hostport(dns_server, (struct sockaddr_in *)&sockaddr,
+                                 false /* log_errors */)) {
+      r->dns_server_addr.family = AF_INET;
+      struct sockaddr_in *in = (struct sockaddr_in *)&sockaddr;
+      memcpy(&r->dns_server_addr.addr.addr4, &in->sin_addr,
+             sizeof(struct in_addr));
+      r->dns_server_addr.tcp_port = ntohs(in->sin_port);
+      r->dns_server_addr.udp_port = ntohs(in->sin_port);
+    } else if (grpc_parse_ipv6_hostport(dns_server,
+                                        (struct sockaddr_in6 *)&sockaddr,
+                                        false /* log_errors */)) {
+      r->dns_server_addr.family = AF_INET6;
+      struct sockaddr_in6 *in6 = (struct sockaddr_in6 *)&sockaddr;
+      memcpy(&r->dns_server_addr.addr.addr6, &in6->sin6_addr,
+             sizeof(struct in6_addr));
+      r->dns_server_addr.tcp_port = ntohs(in6->sin6_port);
+      r->dns_server_addr.udp_port = ntohs(in6->sin6_port);
+    } else {
+      error = grpc_error_set_str(
+          GRPC_ERROR_CREATE_FROM_STATIC_STRING("cannot parse authority"),
+          GRPC_ERROR_STR_TARGET_ADDRESS, grpc_slice_from_copied_string(name));
+      goto error_cleanup;
+    }
+    int status = ares_set_servers_ports(*channel, &r->dns_server_addr);
```

Comment:
We just removed c-ares from node grpc in v1.3.x branch (#10744). This should not be a problem anymore. Please wait until the upmerge PR (#10770) gets checked in.